### DISCUSSION TITLE
where did the `container` go in v4?

### DISCUSSION THREAD

**Discussion by MrOxMasTer** [tailwindlabs/tailwindcss#14801] [General]
where did the `container` go in v4? Or has it been replaced by the CSS container `@container`?

theme: {
    container: {
      center: true,
      padding: {
        DEFAULT: '0.5rem',
        sm: '0.5rem',
        mm: '1rem',
        mt: '4rem',
        md: '5rem',
      },
    },
}

**Comment by wongjn** [comment]
There is an open issue regarding this, see #13129. Also of note, there was a PR to add it into v4, but was closed without merge with reasoning at https://github.com/tailwindlabs/tailwindcss/pull/13391#issuecomment-2025796769.

As an aside, `@container` is not analagous to `container`.

**Comment by aniruddhaadak80** [comment]
Hi @MrOxMasTer ,

In Tailwind CSS v4, the concept of the `container` utility has been enhanced with support for CSS containers. While the traditional `container` utility is still available, you can now utilize the new CSS container feature by using the `@container` directive for more responsive designs.

Here's how you can define a container in your `tailwind.config.js`:

```javascript
theme: {
  container: {
    center: true,
    padding: {
      DEFAULT: '0.5rem',
      sm: '0.5rem',
      mm: '1rem',
      mt: '4rem',
      md: '5rem',
    },
  },
}
```

You can still use the `container` utility as before, and the new CSS `@container` feature allows for more advanced styling. If you need to replace the utility with the CSS approach, ensure to check the [official documentation](https://tailwindcss.com/docs/container) for detailed usage.

Let me know if you have any further questions!

Thanks .

**Reply by adamwathan** [reply]
Just flagging this as incorrect so no one gets confused — `container` is not in v4 currently, the configuration example above is not currently supported, and `@container` is a completely unrelated feature.

**Reply by mofodox** [reply]
Hi @adamwathan , just wanted to check if we are still possible to do the same container configuration in v4?

**Reply by wongjn** [reply]
Yep, see https://tailwindcss.com/docs/upgrade-guide#container-configuration

**Reply by gribbl** [reply]
And how do we define padding and container width based on breakpoints?

**Reply by wongjn** [reply]
You'd write CSS for it within a `@utility` like:

```css
@utility container {
  padding-inline: 1rem;

  @variant sm {
    …
  }

  @variant md {
    …
  }
}
```

**Reply by ayan878** [reply]
By using utility class it changes margin and padding inside the div where the container is define but doest effect container itself

@utility container {
  margin-inline: auto;
  padding-inline: 2rem;
}

**Reply by devSubham35** [reply]
Use this to resize the container - "container max-w-6xl"
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/0d99ef1f-435c-44a2-8364-b272d1053a62" />

**Comment by umkasanki** [comment]
Could someone clarify how to disable `container` in the TailwindCSS v4?
Thanks

**Reply by wongjn** [reply]
Add the following to your CSS:

```css
@source not inline("container");
```

**Reply by zipper** [reply]
@wongjn But then you can’t use this utility name (or, more generally, the class name container) anywhere - even in other layers - which is rather unfortunate 🙁. While this doesn’t matter for 99% of utility names (since no one usually wants to change the behavior of something like flex), the name container isn’t actually tied to a specific CSS property, which breaks the utility paradigm a bit. I’ve always thought of it more as a simple component (or better yet, a "layout class") so it would make sense to be able to opt out of the default Tailwind definition of this component, as was possible in Tailwind v3.

**Reply by wongjn** [reply]
You could consider forking the project to suit your needs.

**Comment by smith558** [comment]
Why is it not in the docs anymore if it's still supported?

**Reply by wongjn** [reply]
It is mentioned in [the upgrade guide](https://tailwindcss.com/docs/upgrade-guide#container-configuration) at least. Though generally, if something is not in the docs, it is considered deprecated.

**Reply by smith558** [reply]
I've found it now, it's a bit hidden [here](https://tailwindcss.com/docs/max-width#using-breakpoints-container) (https://tailwindcss.com/docs/max-width#using-breakpoints-container). Doesn't mention anything about deprecation, though. 🤔

**Reply by wongjn** [reply]
Ah, well found! Yep, so it ***is*** in the docs so not considered deprecated 👍

**Reply by devSubham35** [reply]
Use this to resize the container - "container max-w-6xl"
